pkg/controller/bastion/options.go

```diff
@@ -79,19 +83,31 @@ func DetermineOptions(bastion *extensionsv1alpha1.Bastion, cluster *controller.C
 		"Name": &baseResourceName,
 		"Type": to.StringPtr("gardenctl"),
 	}
-
+	infrastructureConfig := &api.InfrastructureConfig{}
+	if err = json.Unmarshal(cluster.Shoot.Spec.Provider.InfrastructureConfig.Raw, infrastructureConfig); err != nil {
+		return nil, err
+	}
+	// own vnet enabled if own vnet resourceGroup exist
+	myVnetEnabled := false
+	vNetrg := ""
+	if infrastructureConfig.Networks.VNet.ResourceGroup != nil && *infrastructureConfig.Networks.VNet.ResourceGroup != "" {
+		vNetrg = *infrastructureConfig.Networks.VNet.ResourceGroup
+		myVnetEnabled = true
+	}
 	return &Options{
-		BastionInstanceName: baseResourceName,
-		BastionPublicIPName: publicIPResourceName(baseResourceName),
-		SecretReference:     secretReference,
-		CIDRs:               cidrs,
-		WorkersCIDR:         workersCidr,
-		DiskName:            DiskResourceName(baseResourceName),
-		Location:            cluster.Shoot.Spec.Region,
-		ResourceGroupName:   resourceGroup,
-		NicName:             NicResourceName(baseResourceName),
-		Tags:                tags,
-		SecurityGroupName:   NSGName(clusterName),
+		BastionInstanceName:     baseResourceName,
+		BastionPublicIPName:     publicIPResourceName(baseResourceName),
+		SecretReference:         secretReference,
+		CIDRs:                   cidrs,
+		WorkersCIDR:             workersCidr,
+		DiskName:                DiskResourceName(baseResourceName),
+		Location:                cluster.Shoot.Spec.Region,
+		ResourceGroupName:       resourceGroup,
+		MyVnetEnabled:           myVnetEnabled,
+		MyVnetResourceGroupName: vNetrg,
```

Comment:
I don't think that `MyVnetEnabled` is that useful as a field. I would do the following: 

Remove `MyVnetEnabled`

and

Rename `MyVnetResourceGroupName` -> `VnetResourceGroupName`. You can always calculate it when you calculate the `opt` and use that instead of introducing the conditionals in deep nested functions like `getSubnet`. 

You can alternatively, *not* introduce these 2 fields and perform the calculation in a place where you have the status, e.g. pass it with the other args in `ensureNic` [like here](https://github.com/gardener/gardener-extension-provider-azure/blob/d31fce706f474b8b1fe211dc22a882f3ba23a762/pkg/controller/bastion/actuator_reconcile.go#L74)

Reply:
done remove 2 fields and pass it in `ensureNic` bdd779563772700d8883ace9ceb911f121f52817